sympy/series/sequences.py

```diff
@@ -0,0 +1,1123 @@
+            function = Lambda(v1, func1.expr + func2.expr.subs(v2, v1))
+            return SeqFunc(function, self._intersect_interval(other))
+
+    def _mul(self, other):
+        """See docstring of SeqBase._mul"""
+        if isinstance(other, SeqFunc)
+            func1, v1 = self.function, self.variables[0]
+            func2, v2 = other.function, other.variables[0]
+            function = Lambda(v1, func1.expr * func2.expr.subs(v2, v1))
+            return SeqFunc(function, self._intersect_interval(other))
+
+    def coeff_mul(self, coeff):
+        """See docstring of SeqBase.coeff_mul"""
+        coeff = sympify(coeff)
+        expr = self.function.expr * coeff
+        return SeqFunc(Lambda(self.variables[0], expr), \
+                          (self.interval, self.step))
+
+
+def sequence(**kwargs):
+    """Returns appropriate sequence object.
+
+    Examples
+    ========
+
+    >>> from sympy import sequence, SeqPer, SeqFunc, SeqFormula, Lambda
+    >>> from sympy.abc import n
+
+    >>> sequence(formula=(n**2, n), interval=(0, 5))
+    SeqFormula((n**2, n), ([0, 5], 1))
+
+    >>> sequence(periodical=(1, 2, 3), interval=(0, 5))
+    SeqPer((1, 2, 3), ([0, 5], 1))
+
+    >>> sequence(func=Lambda(n, n**2), interval=(0, 5))
+    SeqFunc(Lambda(n, n**2), ([0, 5], 1))
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqPer
+    sympy.series.sequences.SeqFormula
+    sympy.series.sequences.SeqFunc
+    """
+    interval = kwargs.pop('interval', None)
+    if interval is None:
+        interval = (0, S.Infinity, 1)
+
+    key = kwargs.pop('periodical', None)
+    if key is not None:
+        return SeqPer(key, interval)
+
+    key = kwargs.pop('func', None)
+    if key is not None:
+        return SeqFunc(key, interval)
+
+    key = kwargs.pop('formula', None)
+    if key is not None:
+        return SeqFormula(key, interval)
+
+    raise ValueError('Invalid Arguments')
+
+
+################################################################################
+#                            OPERATIONS                                        #
+################################################################################
+
+
+class SeqExprOp(SeqBase):
+    """Base class for operations on sequences
+
+    Examples
+    ========
+
+    >>> from sympy.series.sequences import SeqExprOp, sequence
+    >>> from sympy.abc import n
+    >>> s1 = sequence(formula=(n**2, n), interval=(0, 10))
+    >>> s2 = sequence(periodical=(1, 2, 3), interval=(5, 10))
+    >>> s = SeqExprOp(s1, s2)
+    >>> s.gen
+    ((n**2, n), (1, 2, 3))
+    >>> s.interval
+    [5, 10]
+    >>> s.length
+    6
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqAdd
+    sympy.series.sequences.SeqMul
+    """
+    @property
+    def gen(self):
+        """Generator for the sequence
+        returns a tuple of generators of all the argument sequences
+        """
+        return tuple([a.gen for a in self.args])
```

Comment:
`tuple(a.gen for a in self.args)`. No need to create a list, and then put it in a `tuple`.